src/main/java/clients/TestResultsClient.java

```diff
@@ -50,6 +50,15 @@ public Response getTestResultsSysNumber(String systemNumber) {
             saveUtils();
             response = callGetTestResults(systemNumber);
         }
+        // waiting 5 seconds for the data to be processed
```

Comment:
I would do a loop here and try every 500 miliseconds, because like this we increase the length of the test with 5 seconds no matter what, even if after one second it no longer gives 404.

Reply:
this will trigger ONLY IF it receives 404. Based on previous runs, it only happens ones per run, if ever. So a sofisticated retry mechanism was uncalled for. 
(Will change if it turns out it triggers more than once.)